framework/db/BaseActiveRecord.php

```diff
@@ -419,7 +418,11 @@ public function getRelatedRecords()
      */
     public function hasAttribute($name)
     {
-        return isset($this->_attributes[$name]) || in_array($name, $this->attributes());
+        if (!(isset($this->_attributes[$name]) || array_key_exists($name, $this->_attributes))) {
```

Comment:
As far as I can tell it does the same (not the condition but the whole function).
Make a test if you doubt it, though ;-) I'm fairly sure this function will pass the test same as master would.